### ISSUE TITLE
Something happen in 1.2.1? 

### ISSUE THREAD

**Issue by braginteractive** [edmundhung/conform#780]
### Describe the bug and the expected behavior

I cant really figure out what is going on after installing 1.2.1? Sometimes my forms work as they should, sometimes it takes several clicks on the submit button for things to work properly. 

I rolled back to the previous version I was using and everything works as it should. 

Wish I could help determine what is going on.. 

### Conform version

v1.2.1

### Steps to Reproduce the Bug or Issue

1. useForm

```
	const [form, fields] = useForm({
		id: 'create-task',
		constraint: getZodConstraint(TicketTaskSchema),
		onValidate({ formData }) {
			return parseWithZod(formData, {
				schema: TicketTaskSchema,
			})
		},
		shouldRevalidate: 'onBlur',
	})
```

2. Create a form with the values
3. Try to submit with button
4. No errors
5. Click Submit button again, sometimes it submits, sometimes it doesnt
6. I dunno.

Rolling back to the previous version for now. 

### What browsers are you seeing the problem on?

Chrome

**Comment by edmundhung** [comment]
Yes. I hope v1.2.2 resolves the issue you had. I am sorry for the trouble. Just let me know if you are still running into any issues :)

**Comment by braginteractive** [comment]
Mostly looking to get the key spread error fixed. Thanks
